airbyte-integrations/connectors/source-jina-ai/source_jina_ai/manifest.yaml

```diff
@@ -0,0 +1,158 @@
+      schema:
+        $ref: "#/schemas/reader"
+
+  search_stream:
+    type: DeclarativeStream
+    retriever:
+      type: SimpleRetriever
+      record_selector:
+        $ref: "#/definitions/selector"
+      paginator:
+        type: NoPagination
+      requester:
+        type: HttpRequester
+        url_base: "https://s.jina.ai/{{ config['search_prompt'] }}"
+        http_method: "GET"
+        path: "/"
+        authenticator:
+          type: BearerAuthenticator
+          api_token: "{{ config['api_key'] }}"
+        request_headers:
+          Accept: application/json
+          X-With-Links-Summary: "{{ config['gather_links']}}"
+          X-With-Images-Summary: "{{ config['gather_images']}}"
+    name: "search"
+    $parameters:
+      path: "/example"
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        $ref: "#/schemas/search"
+
+streams:
+  - "#/definitions/reader_stream"
+  - "#/definitions/search_stream"
+
+check:
+  type: CheckStream
+  stream_names:
+    - "reader"
+    - "search"
+
+schemas:
+  search:
+    $schema: http://json-schema.org/draft-07/schema#
+    title: Jina AI URL Reader Schema
+    type: object
+    additionalProperties: true
+    properties:
+      title:
+        type: string
+      url:
+        type: string
+      content:
+        type: string
+      description:
+        type: string
+      links:
+        type: object
+        additionalProperties: true
+        properties:
+          More information...:
+            type: string
+
+  reader:
+    $schema: http://json-schema.org/draft-07/schema#
+    title: Jina AI prompt Reader Schema
+    type: object
+    additionalProperties: true
+    properties:
+      title:
+        type: string
+      url:
+        type: string
+      content:
+        type: string
+      description:
+        type: string
+      links:
+        type: object
+        additionalProperties: true
+        properties:
+          More information...:
+            type: string
+
+spec:
+  type: Spec
+  documentation_url: https://docs.airbyte.com/integrations/sources/jina-ai
+  connection_specification:
+    title: Jina Ai Spec
+    type: object
+    required:
+      - api_key
+    additionalProperties: true
+    properties:
+      api_key:
+        type: string
+        title: API Key
+        description: >-
+          Your Jina AI API Key. See <a
+          href="https://jina.ai/reader/#apiform">here</a>.
+        airbyte_secret: true
+        order: 0
+      read_prompt:
+        type: string
+        title: Your read prompt url, that would be appended at the end of reader stream
+        default: "https://www.google.com"
+        order: 1
+      search_prompt:
+        type: string
+        title: >-
+          Your search prompt, that would be appended at the end of search stream, Should be an encoded string without spaces.
+          Visit https://coding.tools/url-encode for seamless encoding
+        default: "Search%20airbyte"
+        order: 2
```

Comment:
I'm less familiar with this endpoint, but I was thinking similarly as above: perhaps we could make plural and accept `search_prompts` as an array of search strings?

Also - from a usability perspective (maybe also for security/stability), what do you think of requesting an unencoded string like "who is the current president?" and resolving (internally) to something like "who%20is%20the%20current%20president"?

Reply:
Same as URL, it can't accept multiple search prompts I guess
we could use the connector with multiple configs in a loop with pyairbyte for making the affect :)